### PR TITLE
Show testcase variants in UI.

### PR DESCRIPTION
**Pull request by inferno-chromium**

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Deployed on internal chromium staging, can check testcase like 4819226810777600

File: src/appengine/private/components/testcase-detail/testcase-detail.html [hunk 1]
```diff
@@ -596,6 +597,8 @@
               testcase-id="[[info.testcase.id]]"></crash-stats>
         </template>
 
+        <testcase-variants info="[[info]]"></testcase-variants>
```
Comment (oliverchang) [inline src/appengine/private/components/testcase-detail/testcase-detail.html]: only testcase id is used here. better to just pass testcase-id="[[info.testcase.id]]" ?

File: src/appengine/handlers/testcase_detail/testcase_variants.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+"""Handler for getting testcase variants."""
+from datastore import data_types
+from handlers import base_handler
+from libs import handler
+
+
+class Handler(base_handler.Handler):
+  """Handler that return testcase variants."""
+
+  @staticmethod
```
[the remark this answers is not in the document]
Comment (inferno-chromium) [reply src/appengine/handlers/testcase_detail/testcase_variants.py]: Done!

File: src/appengine/handlers/cron/triage.py [hunk 3]
```diff
@@ -42,6 +42,7 @@
 
 def _add_triage_message(testcase, message):
   """Add a triage message."""
+  testcase = data_handler.get_testcase_by_id(testcase.key.id())
```
[the remark this answers is not in the document]
Comment (inferno-chromium) [reply src/appengine/handlers/cron/triage.py]: Done!

File: src/appengine/private/components/testcase-detail/testcase-detail.html (hunk 1 above)
Comment (inferno-chromium) [reply src/appengine/private/components/testcase-detail/testcase-detail.html]: Done!

File: src/python/datastore/data_handler.py [hunk 4]
```diff
@@ -1245,6 +1245,23 @@ def get_entity_by_type_and_id(entity_type, entity_id):
   return entity_type.get_by_id(int(entity_id))
 
 
+# ------------------------------------------------------------------------------
+# TestcaseVariant related functions
+# ------------------------------------------------------------------------------
+
+
+def get_testcase_variant_entity(testcase_id, job_type):
```
[the remark this answers is not in the document]
Comment (inferno-chromium) [reply src/python/datastore/data_handler.py]: Done!

**Pull request merged**